yt_dlp/extractor/cda.py

```diff
@@ -173,18 +175,36 @@ def extract_format(page, version):
                     video['file'] = video['file'].replace('adc.mp4', '.mp4')
             elif not video['file'].startswith('http'):
                 video['file'] = decrypt_file(video['file'])
-            f = {
+            c_quality = video.get('quality')
+            qualities = video.get('qualities')
+            for q in qualities:
+                if qualities[q] == c_quality:
+                    c_quality = q
+            info_dict['formats'].append({
                 'url': video['file'],
-            }
-            m = re.search(
-                r'<a[^>]+data-quality="(?P<format_id>[^"]+)"[^>]+href="[^"]+"[^>]+class="[^"]*quality-btn-active[^"]*">(?P<height>[0-9]+)p',
-                page)
-            if m:
-                f.update({
-                    'format_id': m.group('format_id'),
-                    'height': int(m.group('height')),
-                })
-            info_dict['formats'].append(f)
+                'format_id': c_quality,
+                'height': int_or_none(c_quality[:-1]),
+            })
+            if qualities:
+                for quality in qualities:
```

Comment:
```suggestion
                for quality, value in qualities.items():
```
then u can use `value` instead of `qualities[quality]` inside the loop
PS: dont literally name the variable `value`